### ISSUE TITLE
Cannot Connect to Audible and app hangs after clicking Help (or F1)

### ISSUE THREAD

**Issue by tomsem** [openaudible/openaudible#1114]
Things seem to have been working a week or so ago. 

I tried Quick Sync, this failed with some permissions error (I think).
Then tried Connect To Audible, it failed. I checked license, just updated, applied new license key. I noticed complaint that OpenAudible/web folder was not found, so I created an empty one.

(May be some macOS security update has broken something) running Ventura 13.0.1 (22A400), getting 13.2 to see if it fixes anything.

But it is still failing. Here is log:

com.teamdev.jxbrowser.engine.ChromiumBinariesExtractionException: Failed to extract Chromium binaries into /Users/thomas/Library/OpenAudible/tmp/jxbrowser
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:59)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.lambda$extract$0(ChromiumExtractor.java:38)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.extract(ChromiumExtractor.java:35)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.extractChromiumBinariesIfNecessary(EngineImpl.java:422)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.newInstance(EngineImpl.java:149)
	at com.teamdev.jxbrowser.engine.Engine.newInstance(Engine.java:163)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.<init>(JxBrowser.java:72)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.create(JxBrowser.java:58)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.lambda$createJxBrowser$0(BrowserFactory.java:116)
	at org.openaudible.desktop.swt.gui.SWTAsync$1.task(SWTAsync.java:172)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:292)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:183)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:147)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:175)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createJxBrowser(BrowserFactory.java:115)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createBrowserShell(BrowserFactory.java:94)
	at org.openaudible.desktop.swt.manager.browser.BrowserManager.get(BrowserManager.java:31)
	at org.openaudible.audible.APIManager.browse(APIManager.java:238)
	at org.openaudible.audible.APIManager.login(APIManager.java:546)
	at org.openaudible.desktop.swt.manager.AudibleGUI$7.run(AudibleGUI.java:1130)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connectToAudibleAPI(AudibleGUI.java:1137)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connect(AudibleGUI.java:317)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.execute(CommandCenter.java:339)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.handleMenuAction(CommandCenter.java:65)
	at org.openaudible.desktop.swt.manager.menu.AppMenu.widgetSelected(AppMenu.java:531)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4577)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1547)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1532)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1325)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4363)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3939)
	at org.openaudible.desktop.swt.gui.GUI.runEventLoop(GUI.java:291)
	at org.openaudible.desktop.swt.gui.GUI.run(GUI.java:391)
	at org.openaudible.desktop.swt.manager.AppLoader.<init>(AppLoader.java:137)
	at org.openaudible.desktop.swt.manager.AppLoader.main(AppLoader.java:150)
	at org.openaudible.desktop.Application.main(Application.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.exe4j.runtime.LauncherEngine.launch(LauncherEngine.java:84)
	at com.install4j.runtime.launcher.MacLauncher.main(MacLauncher.java:52)
Caused by: java.io.IOException: Cannot run program "/Users/thomas/Library/OpenAudible/tmp/jxbrowser/7zr-mac": error=13, Permission denied
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at java.base/java.lang.Runtime.exec(Runtime.java:592)
	at java.base/java.lang.Runtime.exec(Runtime.java:451)
	at com.teamdev.jxbrowser.internal.xz.SevenZip.decompress(SevenZip.java:91)
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:56)
	... 45 more
Caused by: java.io.IOException: error=13, Permission denied
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 50 more

Jan 24, 2023 1:32:32 AM org.openaudible.audible.APIManager browse
WARNING: Serious error getting browser... trying one more time...
Jan 24, 2023 1:32:32 AM org.openaudible.desktop.swt.manager.browser.BrowserFactory createBrowserShell
INFO: Using jxbrowser. This is experimental and using a demo license that will expire!
Jan 24, 2023 1:32:32 AM org.openaudible.desktop.swt.gui.GUI run
SEVERE: Error running GUI task block GUITask' [2692]
com.teamdev.jxbrowser.engine.ChromiumBinariesExtractionException: Failed to extract Chromium binaries into /Users/thomas/Library/OpenAudible/tmp/jxbrowser
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:59)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.lambda$extract$0(ChromiumExtractor.java:38)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.extract(ChromiumExtractor.java:35)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.extractChromiumBinariesIfNecessary(EngineImpl.java:422)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.newInstance(EngineImpl.java:149)
	at com.teamdev.jxbrowser.engine.Engine.newInstance(Engine.java:163)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.<init>(JxBrowser.java:72)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.create(JxBrowser.java:58)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.lambda$createJxBrowser$0(BrowserFactory.java:116)
	at org.openaudible.desktop.swt.gui.SWTAsync$1.task(SWTAsync.java:172)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:292)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:183)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:147)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:175)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createJxBrowser(BrowserFactory.java:115)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createBrowserShell(BrowserFactory.java:94)
	at org.openaudible.desktop.swt.manager.browser.BrowserManager.get(BrowserManager.java:31)
	at org.openaudible.audible.APIManager.browse(APIManager.java:241)
	at org.openaudible.audible.APIManager.login(APIManager.java:546)
	at org.openaudible.desktop.swt.manager.AudibleGUI$7.run(AudibleGUI.java:1130)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connectToAudibleAPI(AudibleGUI.java:1137)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connect(AudibleGUI.java:317)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.execute(CommandCenter.java:339)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.handleMenuAction(CommandCenter.java:65)
	at org.openaudible.desktop.swt.manager.menu.AppMenu.widgetSelected(AppMenu.java:531)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4577)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1547)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1532)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1325)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4363)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3939)
	at org.openaudible.desktop.swt.gui.GUI.runEventLoop(GUI.java:291)
	at org.openaudible.desktop.swt.gui.GUI.run(GUI.java:391)
	at org.openaudible.desktop.swt.manager.AppLoader.<init>(AppLoader.java:137)
	at org.openaudible.desktop.swt.manager.AppLoader.main(AppLoader.java:150)
	at org.openaudible.desktop.Application.main(Application.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.exe4j.runtime.LauncherEngine.launch(LauncherEngine.java:84)
	at com.install4j.runtime.launcher.MacLauncher.main(MacLauncher.java:52)
Caused by: java.io.IOException: Cannot run program "/Users/thomas/Library/OpenAudible/tmp/jxbrowser/7zr-mac": error=13, Permission denied
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at java.base/java.lang.Runtime.exec(Runtime.java:592)
	at java.base/java.lang.Runtime.exec(Runtime.java:451)
	at com.teamdev.jxbrowser.internal.xz.SevenZip.decompress(SevenZip.java:91)
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:56)
	... 45 more
Caused by: java.io.IOException: error=13, Permission denied
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 50 more

Jan 24, 2023 1:32:32 AM org.openaudible.desktop.swt.manager.browser.BrowserFactory createBrowserShell
INFO: Using jxbrowser. This is experimental and using a demo license that will expire!
Jan 24, 2023 1:32:32 AM org.openaudible.desktop.swt.gui.GUI run
SEVERE: Error running GUI task block GUITask' [5413]
com.teamdev.jxbrowser.engine.ChromiumBinariesExtractionException: Failed to extract Chromium binaries into /Users/thomas/Library/OpenAudible/tmp/jxbrowser
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:59)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.lambda$extract$0(ChromiumExtractor.java:38)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.teamdev.jxbrowser.internal.ChromiumExtractor.extract(ChromiumExtractor.java:35)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.extractChromiumBinariesIfNecessary(EngineImpl.java:422)
	at com.teamdev.jxbrowser.engine.internal.EngineImpl.newInstance(EngineImpl.java:149)
	at com.teamdev.jxbrowser.engine.Engine.newInstance(Engine.java:163)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.<init>(JxBrowser.java:72)
	at org.openaudible.desktop.swt.manager.browser.jxbrowser.JxBrowser.create(JxBrowser.java:58)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.lambda$createJxBrowser$0(BrowserFactory.java:116)
	at org.openaudible.desktop.swt.gui.SWTAsync$1.task(SWTAsync.java:172)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:292)
	at org.openaudible.desktop.swt.gui.SWTAsync.run(SWTAsync.java:183)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:147)
	at org.openaudible.desktop.swt.gui.SWTAsync.block(SWTAsync.java:175)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createJxBrowser(BrowserFactory.java:115)
	at org.openaudible.desktop.swt.manager.browser.BrowserFactory.createBrowserShell(BrowserFactory.java:94)
	at org.openaudible.desktop.swt.manager.browser.BrowserManager.get(BrowserManager.java:31)
	at org.openaudible.desktop.swt.manager.browser.BrowserManager.get(BrowserManager.java:37)
	at org.openaudible.desktop.swt.manager.browser.BrowserManager.setURL(BrowserManager.java:127)
	at org.openaudible.audible.APIManager.browse(APIManager.java:244)
	at org.openaudible.audible.APIManager.login(APIManager.java:546)
	at org.openaudible.desktop.swt.manager.AudibleGUI$7.run(AudibleGUI.java:1130)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connectToAudibleAPI(AudibleGUI.java:1137)
	at org.openaudible.desktop.swt.manager.AudibleGUI.connect(AudibleGUI.java:317)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.execute(CommandCenter.java:339)
	at org.openaudible.desktop.swt.manager.menu.CommandCenter.handleMenuAction(CommandCenter.java:65)
	at org.openaudible.desktop.swt.manager.menu.AppMenu.widgetSelected(AppMenu.java:531)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4577)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1547)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1532)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1325)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4363)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3939)
	at org.openaudible.desktop.swt.gui.GUI.runEventLoop(GUI.java:291)
	at org.openaudible.desktop.swt.gui.GUI.run(GUI.java:391)
	at org.openaudible.desktop.swt.manager.AppLoader.<init>(AppLoader.java:137)
	at org.openaudible.desktop.swt.manager.AppLoader.main(AppLoader.java:150)
	at org.openaudible.desktop.Application.main(Application.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.exe4j.runtime.LauncherEngine.launch(LauncherEngine.java:84)
	at com.install4j.runtime.launcher.MacLauncher.main(MacLauncher.java:52)
Caused by: java.io.IOException: Cannot run program "/Users/thomas/Library/OpenAudible/tmp/jxbrowser/7zr-mac": error=13, Permission denied
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at java.base/java.lang.Runtime.exec(Runtime.java:592)
	at java.base/java.lang.Runtime.exec(Runtime.java:451)
	at com.teamdev.jxbrowser.internal.xz.SevenZip.decompress(SevenZip.java:91)
	at com.teamdev.jxbrowser.internal.xz.XzExtractor.extract(XzExtractor.java:56)
	... 47 more
Caused by: java.io.IOException: error=13, Permission denied
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
	... 52 more

**Comment by openaudible** [comment]
OK. That does seem like a firewall issue or something hitting the jxbrowser code... which we have removed in the latest beta.
Can you try the beta, which has two login methods to choose from: https://openaudible.org/beta

**Comment by tomsem** [comment]
I have installed the beta.

Unfortunately I cannot login with either of the two methods.

The Recommended method asks for Audible User Name and Password. I filled those in (tried both User Name and Email), but Next button is not enabled so I cannot continue.

The Alternate method gives error registering/creating virtual device. I have only 3 devices registered, it's not at all obvious deleting any of these will help.

**Comment by openaudible** [comment]
The log in dialog needs a user/password before the next button will activate. That has been changed for the next release.
The too many devices might need a look. That is rare these days-- maybe another bug has triggered that. 
You can look at the number of audible devices by going to your amazon account and look at Account, manage content and devices, audible devices.. 
Can you try again with the password.. and or send a bug report from the About menu.
Sorry for the late reply--failed to hit send.

**Comment by tomsem** [comment]
Next button is not activated even though both user and password are there.

Again, only 3 devices are activated.

For now I am downloading manually and importing.

Another weird bug: when theme is Dark, Bug Report does not work (no way to enter email or Issue #).

**Comment by tomsem** [comment]
I have installed 3.6.3.11beta, and was eventually able to Connect To Audible using the Secure OpenAudible option (the web browser option didn't work, same 'SEVERE: showError: Error registering virtual device' as before).

At first I was getting CAPTHA and couldn't get past it, but at some point it started sending more familiar OTP prompt  and login succeeded. I used my Audible email and not Audible 'User Name', I don't know if the latter even works so maybe the UI prompt should be changed. Closing the issue. Thank you!
